src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -0,0 +1,55 @@
+import { useCallback } from 'react'
+import { useClearQuote, useUpdateQuote } from 'state/price/hooks'
+import { getCanonicalMarket, registerOnWindow } from 'utils/misc'
+import { FeeQuoteParams, getFeeQuote, getPriceQuote } from 'utils/operator'
+
+export interface RefetchQuoteCallbackParmams {
+  quoteParams: FeeQuoteParams
+  fetchFee: boolean
+}
+
+/**
+ * @returns callback that fetches a new quote and update the state
+ */
+export function useRefetchQuoteCallback() {
+  const updateQuote = useUpdateQuote()
+  const clearQuote = useClearQuote()
+  registerOnWindow({ updateQuote })
+
+  return useCallback(
+    async ({ quoteParams, fetchFee }: RefetchQuoteCallbackParmams) => {
+      const { sellToken, buyToken, amount, kind, chainId } = quoteParams
+      const { baseToken, quoteToken } = getCanonicalMarket({ sellToken, buyToken, kind })
+
+      // Get a new price quote
+      const pricePromise = getPriceQuote({ chainId, baseToken, quoteToken, amount, kind })
+
+      // Get a new fee quote (if required)
+      const feePromise = fetchFee
+        ? getFeeQuote({ chainId, sellToken, buyToken, amount, kind }).catch(err => {
+            console.error('Error fetching the fee', err)
+            return null
+          })
+        : Promise.resolve(null)
```

Comment:
can just be `null`or am i missing something? `Promise.all([null])`resolves fine or is this a type issue?